packages/mcp-client/src/McpClient.ts

```diff
@@ -193,11 +191,26 @@ export class McpClient {
 				messages.push(toolMessage);
 				return yield toolMessage;
 			}
+			let toolArgs: Record<string, unknown> = {};
+			try {
+				toolArgs = toolCall.function.arguments === "" ? {} : JSON.parse(toolCall.function.arguments);
+			} catch (error) {
+				if (error instanceof SyntaxError) {
+					toolMessage.content = `Invalid JSON generated by the model: ${error.message}`;
```

Comment:
nit: in the case of a malformed JSON, we can short-circuit the tool execution, inject the error back and continue
```suggestion
					toolMessage.content = `Invalid JSON generated by the model: ${error.message}`;
					messages.push(toolMessage);
					yield toolMessage;
					continue;
```

Reply:
ah yes you are correct,  i intended to do this